### PR TITLE
Fix problems with other plugins' path replacement

### PR DESCRIPTION
**Pull request by fatfisz**
**What**: This PR fixes an issue described in kentcdodds/import-all.macro#7 and similar issues.

**Why**: The plugin may be used besides other plugins, most notably babel-plugin-istanbul, which negatively affects how babel-plugin-macros behaves.

**How**: A hack is used that solves the problem.

First I'll push without the fix just to see the tests fail.
Just after that I'll push the fix.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -124,7 +124,11 @@ function macrosPlugin(babel, {require: _require = require} = {}) {
 
 // eslint-disable-next-line complexity
 function applyMacros({path, imports, source, state, babel, interopRequire}) {
-  const {file: {opts: {filename}}} = state
+  const {
+    file: {
+      opts: {filename},
+    },
+  } = state
```
Comment (fatfisz) [inline src/index.js]: This was done automatically by the linter...

File: src/__tests__/index.js [hunk 2]
```diff
@@ -254,5 +254,16 @@ pluginTester({
         }
       },
     },
+    {
```
Comment (kentcdodds) [inline src/__tests__/index.js]: Could you verify for me that this test fails without the changes in the source? Maybe take a screenshot?

Comment (kentcdodds) [reply src/__tests__/index.js]: Never mind! You thought ahead!

> First I'll push without the fix just to see the tests fail.
> Just after that I'll push the fix.

Thanks!

**Comment by kentcdodds** [comment]
This is so awesome. Thank you for your help here! Let me know if you'd like me to file the issue with Babel or if you're planning on doing that.

**Comment by fatfisz** [comment]
I see you already found an issue for this in Babel repo (you had more success than me, I couldn't find it). I think I'll continue from there. Thanks!

**Pull request merged**